src/systemd-sonic-generator/ssg-test.cc

```diff
@@ -0,0 +1,550 @@
+/*------------------------------------------------------------------
+ * ssg-test.cc - systemd-sonic-generator Unit Test
+ *
+ * Initial: Apr 2021
+ *
+ * Copyright (c) 2021 by Cisco Systems, Inc.
+ *------------------------------------------------------------------
+ */
+#include <mutex>
+#include <string>
+#include <sys/stat.h>
+#include <gtest/gtest.h>
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include "systemd-sonic-generator.h"
+
+namespace fs = boost::filesystem;
+
+namespace SSGTest {
+#define IS_MULTI_ASIC(x)  ((x) > 1)
+#define IS_SINGLE_ASIC(x) ((x) <= 1)
+#define NUM_UNIT_FILES 6
+
+/*
+ * This test class uses following directory hierarchy for input and output
+ * data for systemd-sonic-generator.
+ *
+ *    tests/ssg-test/   --- Test data directory
+ *             |
+ *             |---generated_services.conf
+ *             |---machine.conf (systemd-sonic-generator fetch platform from here)
+ *             |---systemd/
+ *             |       |--- *.service (Test unit files are copied from
+ *             |                       tests/testfiles/ to here)
+ *             |----test_platform/  (test platform)
+ *             |       |---asic.conf
+ *             |
+ *             |----generator/   (Output Directory)
+ *
+ */
+const std::string TEST_ROOT_DIR = "tests/ssg-test/";
+const std::string TEST_UNIT_FILE_PREFIX = TEST_ROOT_DIR + "systemd/";
+const std::string TEST_ASIC_CONF_FORMAT = TEST_ROOT_DIR + "%s/asic.conf";
+const std::string TEST_MACHINE_CONF = TEST_ROOT_DIR + "machine.conf";
+
+const std::string TEST_PLATFORM_DIR = TEST_ROOT_DIR + "test_platform/";
+const std::string TEST_ASIC_CONF = TEST_PLATFORM_DIR + "asic.conf";
+
+const std::string TEST_OUTPUT_DIR = TEST_ROOT_DIR + "generator/";
+
+const std::string TEST_CONFIG_FILE = TEST_ROOT_DIR + "generated_services.conf";
+
+const std::string TEST_UNIT_FILES = "tests/testfiles/";
+
+/* Input data for generated_services.conf */
+const std::vector<std::string> generated_services = {
+    "multi_inst_a.service",     /* Single instance of a  multi asic service a */
+    "multi_inst_a@.service",    /* Multi-instance of a multi asic service a */
+    "multi_inst_b@.service",    /* Multi-instance of a multi asic service b */
+    "single_inst.service",      /* A single instance service */
+    "test.service",             /* A single instance test service
+                                   to test dependency creation */
+    "test.timer",               /* A timer service */
+};
+
+static std::mutex g_ssg_test_mutex;
+
+class SystemdSonicGeneratorFixture: public testing::Test {
```

Comment:
Suggest to add a space before first `:` to align style with other SONiC C++ files.